examples/embeddable_examples/public/react_embeddables/field_list/field_list_embeddable.tsx

```diff
@@ -0,0 +1,243 @@
+) => {
+  const fieldListEmbeddableFactory: EmbeddableFactory<FieldListSerializedState, FieldListApi> = {
+    type: FIELD_LIST_ID,
+    buildEmbeddable: async ({ initialState, finalizeApi, parentApi, uuid }) => {
+      const state = await deserializeState(dataViews, initialState);
+      const allDataViews = await dataViews.getIdsWithTitle();
+      const subscriptions = new Subscription();
+      const titleManager = initializeTitleManager(initialState?.rawState ?? {});
+      const fieldListStateManager = initializeStateManager(state, defaultFieldListState);
+
+      // Whenever the data view changes, we want to update the data views and reset the selectedFields in the field list state manager.
+      subscriptions.add(
+        fieldListStateManager.api.dataViewId$
+          .pipe(
+            skip(1),
+            distinctUntilChanged(),
+            tap(() => fieldListStateManager.api.setSelectedFieldNames([])),
+            switchMap((dataViewId) =>
+              dataViewId ? dataViews.get(dataViewId) : dataViews.getDefaultDataView()
+            )
+          )
+          .subscribe((nextSelectedDataView) => {
+            fieldListStateManager.api.setDataViews(
+              nextSelectedDataView ? [nextSelectedDataView] : undefined
+            );
+          })
+      );
+
+      function serializeState() {
+        const { dataViewId, selectedFieldNames } = fieldListStateManager.getLatestState();
+        const references: Reference[] = dataViewId
+          ? [
+              {
+                type: DATA_VIEW_SAVED_OBJECT_TYPE,
+                name: FIELD_LIST_DATA_VIEW_REF_NAME,
+                id: dataViewId,
+              },
+            ]
+          : [];
+        return {
+          rawState: {
+            ...titleManager.getLatestState(),
+            // here we skip serializing the dataViewId, because the reference contains that information.
+            selectedFieldNames,
+          },
+          references,
+        };
+      }
+
+      const unsavedChangesApi = initializeUnsavedChanges({
+        uuid,
+        parentApi,
+        serializeState,
+        anyStateChange$: merge(titleManager.anyStateChange$, fieldListStateManager.anyStateChange$),
+        getComparators: () => ({
+          ...titleComparators,
+          selectedFieldNames: (a, b) => {
+            return (a?.slice().sort().join(',') ?? '') === (b?.slice().sort().join(',') ?? '');
+          },
+        }),
+        onReset: async (lastSaved) => {
+          const lastState = await deserializeState(dataViews, lastSaved);
+          fieldListStateManager.reinitializeState(lastState);
+          titleManager.reinitializeState(lastSaved?.rawState);
+        },
+      });
+
+      const api = finalizeApi({
+        ...titleManager.api,
+        ...unsavedChangesApi,
+        serializeState,
+      });
+
+      return {
+        api,
+        Component: () => {
+          const selectedFieldNames = useStateFromPublishingSubject(
```

Comment:
How come you are not using `useBatchedPublishingSubjects`?

Reply:
No reason. Changed in [8a485b1](https://github.com/elastic/kibana/pull/218174/commits/8a485b1c70c5c6b7978cdbf5af4edd4dcd5cd60f)